src/main/java/org/mafagafogigante/dungeon/game/PartOfDay.java

```diff
@@ -69,6 +69,24 @@ public static PartOfDay getCorrespondingConstant(Date date) {
     throw new AssertionError(); // Execution should never reach this line.
   }
 
+  /**
+   * Returns the PartOfDay constant corresponding to a given name.
+   *
+   * @param name a name corresponding to PartOfDay enum.
+   * @return a PartOfDay constant.
+   */
+  @NotNull
+  public static PartOfDay getPartOfDayByName(String name) {
+    PartOfDay[] podArray = values();
+    // Note that the array is sorted in ascending startingHour order, therefore we iterate backwards.
```

Comment:
I don't follow: why do we need to iterate in a specific order?

Reply:
Oh I was copying the function from `getCorrespondingConstant`, I can get rid of that comment. 

Thinking about it maybe I could overload the function since it's very similar too `getCorrespondingConstant`